src/api/app/models/user_ldap_strategy.rb

```diff
@@ -355,23 +355,25 @@ def self.initialize_ldap_con(user_name, password)
     nil
   end
 
+  def self.ldap_port
+    return CONFIG['ldap_port'] if CONFIG['ldap_port']
+
+    CONFIG['ldap_ssl'] == :on ? 636 : 389
+  end
+
   def self.try_ldap_con(server, user_name, password)
     # implicitly turn array into string
     user_name = [user_name].flatten.join('')
 
     Rails.logger.debug("Connecting to #{server} as '#{user_name}'")
+    port = ldap_port
+
     begin
-      if CONFIG.has_key?('ldap_ssl') && CONFIG['ldap_ssl'] == :on
-        port = CONFIG.has_key?('ldap_port') ? CONFIG['ldap_port'] : 636
-        conn = LDAP::SSLConn.new(server, port)
+      if CONFIG['ldap_ssl'] == :on || CONFIG['ldap_start_tls'] == :on
```

Comment:
No, I think this is way more readable! Please don't do the nested assignment...
One thing what could improve the readability is moving ``CONFIG['ldap_start_tls'] == :on`` to a variable.